framework/i18n/Formatter.php

```diff
@@ -168,6 +169,21 @@ class Formatter extends Component
      * @see http://php.net/manual/en/class.intlcalendar.php
      * @since 2.0.7
      */
+
+    public $dateRangeFormat = [
+                'showTime' => false,
+                'year' => 'Y',
+                'month' => 'M',
+                'day' => 'j',
+                'time' => 'H:i',
+                'templates' => [
+                    'notDiff' => '{from_time} {from_day} {from_month} {from_year}',
+                    'diffYear' => '{from_time} {from_day} {from_month} {from_year} - {to_time} {to_day} {to_month} {to_year}',
+                    'diffMonth' => '{from_time} {from_day} {from_month} - {to_time} {to_day} {to_month} {to_year}',
+                    'diffDay' => '{from_time} {from_day} - {to_time} {to_day} {to_month} {to_year}',
+                    'diffTime' => '{from_time} - {to_time} {to_day} {to_month} {to_year}'
+                ]
+            ];
```

Comment:
Implementing this properly is not easy, we might take advantage of existing implementations such as https://developer.android.com/reference/android/icu/text/DateIntervalFormat.html
source code: https://github.com/android/platform_external_icu4c/blob/e5311394ca22b280da41cd17059288dab3fb1ea6/i18n/dtitvfmt.cpp